spec/main_spec.rb

```diff
@@ -0,0 +1,234 @@
+# spec :main_spec.rb
+require './main.rb'
+
+RSpec.describe Enumerable do
+  let(:array) { [11, 2, 3, 56] }
+  let(:array_of_strings) { %w[a b c d] }
+  let(:range) { 5..8 }
+
+  describe '.my_each' do
+    context 'when receive a block' do
+      it 'push each value to the result array' do
+        result = []
+        array.my_each { |x| result.push(x) }
+        expect(result).to(eq(array))
+      end
```

Comment:
[Optional]
We want our my_each to work exactly to each method right? So why not check it with it? We can transform the test case something like this->
```ruby
it 'push each value to the result array' do
    expected = []
    array.my_each { |x| result.push(x) }
    current_result = expected.dup
    array.each { |x| result.push(x) }
    expect(current_result).to eq(expected)
end
```